code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/intermediate_model_validation/BuildScriptIntermediateOperationModelValidator.xtend

```diff
@@ -0,0 +1,40 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.intermediate_model_validation
+
+import de.fhdo.lemma.model_processing.annotations.IntermediateModelValidator
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.model_processing.builtin_phases.intermediate_model_validation.AbstractIntermediateDeclarativeValidator
+import de.fhdo.lemma.operation.intermediate.IntermediateContainer
+import org.eclipse.xtext.validation.Check
+
+/**
+ * The build script model validator is responsible for checking the compatibility of the
+ * intermediate operation model with the build script code generation module.
+ *
+ * @author Philip Wizenty(philip.wizenty@fh-dortmund.de)
+ */
+@IntermediateModelValidator
+class BuildScriptIntermediateOperationModelValidator
+    extends AbstractIntermediateDeclarativeValidator {
+
+    /*
+     * Get the namespace for the intermediate operation model
+     */
+    override getLanguageNamespace() {
+        return IntermediatePackage.eNS_URI
+    }
+
+    /*
+     * Check if the deployed microservice uses a maven build technology as the build script
+     * code generation module only supports this technology.
+     */
+    @Check
+    def checkMavenBuildTechnologyForContainer(IntermediateContainer container) {
+        val aspect = container.aspects.findFirst[aspect | aspect.name == "MavenTechnology"]
+        if (aspect.propertyValues.get(0).value == false) {
+            warning("The build script gode generation module only support maven as a build "
```

Comment:
Typos: gode --> code, maven -> Maven

Moreover: The warning does not explain what the actual problems is (wrong build technology) and how to resolve this problem.

Reply:
Also: Why is this a warning? Is it safe to proceed and what will happen if so? (If the decision to this being a warning is correct, then please comment on my questions in the source code for anyone to know.)